minidump-processor/src/op_analysis.rs

```diff
@@ -183,8 +227,214 @@ mod amd64 {
+                | Opcode::RDMSR
+                | Opcode::RDPMC
+                | Opcode::RDTSC
+                | Opcode::RDTSCP
+                | Opcode::RETF
+                | Opcode::STI
+                | Opcode::SWAPGS
+                | Opcode::SYSEXIT
+                | Opcode::SYSRET
+                | Opcode::VMCALL
+                | Opcode::VMCLEAR
+                | Opcode::VMLAUNCH
+                | Opcode::VMPTRLD
+                | Opcode::VMPTRST
+                | Opcode::VMREAD
+                | Opcode::VMRESUME
+                | Opcode::VMWRITE
+                | Opcode::VMXOFF
+                | Opcode::VMXON
+                | Opcode::WBINVD
+                | Opcode::WRMSR
+                | Opcode::XSETBV => true,
+                _ => false,
+            }
+        }
+    }
+
+    #[allow(clippy::upper_case_acronyms)]
+    enum CommonOpcode {
+        ADD,
+        CALL,
+        CMP,
+        DEC,
+        IRET,
+        IRETD,
+        IRETQ,
+        INC,
+        JMP,
+        JMPF,
+        MOV,
+        MOVAPS,
+        MOVUPS,
+        POP,
+        PUSH,
+        RETF,
+        RETURN,
+        SUB,
+    }
+
+    impl CommonOpcode {
+        fn from_amd64_opcode(opcode: Opcode) -> Option<Self> {
+            match opcode {
+                Opcode::ADD => Some(Self::ADD),
+                Opcode::CALL => Some(Self::CALL),
+                Opcode::CMP => Some(Self::CMP),
+                Opcode::DEC => Some(Self::DEC),
+                Opcode::IRET => Some(Self::IRET),
+                Opcode::IRETD => Some(Self::IRETD),
+                Opcode::IRETQ => Some(Self::IRETQ),
+                Opcode::INC => Some(Self::INC),
+                Opcode::JMP => Some(Self::JMP),
+                Opcode::JMPF => Some(Self::JMPF),
+                Opcode::MOV => Some(Self::MOV),
+                Opcode::MOVAPS => Some(Self::MOVAPS),
+                Opcode::MOVUPS => Some(Self::MOVUPS),
+                Opcode::POP => Some(Self::POP),
+                Opcode::PUSH => Some(Self::PUSH),
+                Opcode::RETF => Some(Self::RETF),
+                Opcode::RETURN => Some(Self::RETURN),
+                Opcode::SUB => Some(Self::SUB),
+                _ => None,
+            }
+        }
+
+        fn is_common(opcode: Opcode) -> bool {
+            Self::from_amd64_opcode(opcode).is_some()
+        }
+    }
+
+    impl OperandAccessType {
+        pub fn is_read_or_write(&self) -> bool {
+            !matches!(self, Self::UncommonInstructionAccess)
+        }
+        // TODO: Derive access type using `yaxpeax` instead
+        fn explicit_from_instruction(instruction: Instruction, index: u8) -> Option<Self> {
+            let Some(opcode) = CommonOpcode::from_amd64_opcode(instruction.opcode()) else {
+                return Some(Self::UncommonInstructionAccess);
+            };
+            match opcode {
+                CommonOpcode::ADD | CommonOpcode::SUB => {
+                    if index == 0 {
```

Comment:
Technically this is accounted for? since `OperandAccessType::explicit_from_instruction` was only called (in `GetMemoryAccess::explicit_access`) when it is determined the operand at that index is a memory operand.
But yea the structure is unnecessarily unintuitive. 
Right now it loops through the operands, checks for memory operands, then looks at the opcode and determine the access direction.
A better structure would be to look at the opcode first, and then loop through the operands and determine access types (if any). I'll change it to that.